### PR TITLE
Fixing Aggregated Objective Visualisations

### PR DESCRIPTION
**Pull request by arrjon**
Fixes the isses mentioned in #1410 so `AggregatedObjective` can be used in the `visualize.model_fit` function.

### CODE REVIEW

**Comment by arrjon** [comment]
For the moment I iterate over the objectives in `AggregatedObjectives` to find all amici models and plot them individually. For the case with only one model and a `NegLogPrior` objective this works fine.

However, I did not test the function using a petab problem with multiple amici models. Potentially, one needs to extract information from the petab measurement file specific to the corresponding model.

File: pypesto/visualize/model_fit.py [hunk 1]
```diff
@@ -72,62 +72,106 @@ def visualize_optimized_model_fit(
...
-        simulations_df=simulation_df,
-        **kwargs,
-    )
+    amici_models, objective_results = [], []
 
-    non_quantitative_data_types = get_petab_non_quantitative_data_types(
-        petab_problem
-    )
+    # get amici model
+    if hasattr(pypesto_problem.objective, AMICI_MODEL):
```
Comment (dweindl) [inline pypesto/visualize/model_fit.py:79]: What's the advantage of `hasattr` over `isinstance(. , AmiciObjective)`? The latter seems safer and will make it easier to find usages of `AmiciObjective.amici_model`.

File: pypesto/visualize/model_fit.py [hunk 2]
```diff
@@ -72,62 +72,106 @@ def visualize_optimized_model_fit(
...
-        ):
-            axes = plot_categories_from_pypesto_result(
-                result,
-                start_index=start_index,
-                axes=axes,
+        objective_results.append(
+            pypesto_problem.objective(x, return_dict=True)
+        )
+        amici_models.append(pypesto_problem.objective.amici_model)
+    elif hasattr(pypesto_problem.objective, "_objectives"):
```
Comment (dweindl) [inline pypesto/visualize/model_fit.py:88]: As above - if there is no reason against it, I'd prefer `isinstance`.

File: pypesto/visualize/model_fit.py [hunk 3]
```diff
@@ -72,62 +72,106 @@ def visualize_optimized_model_fit(
...
+        # iterate over objectives to find the amici models in the AggregatedObjective
+        x = result.optimize_result.list[start_index]["x"]
+
+        for objective in pypesto_problem.objective._objectives:
+            if hasattr(objective, AMICI_MODEL):
```
Comment (dweindl) [inline pypesto/visualize/model_fit.py:93]: `isinstance`?

Comment (dweindl) [inline pypesto/visualize/model_fit.py:93]: I'd emit a warning for any non-AmiciObjective encountered here. (Or less preferred add to the documentation that non-AmiciObjective are silently ignored.)

**Review by dweindl — COMMENTED** [review]
Thanks.

File: pypesto/visualize/model_fit.py (hunk 1 above)
Comment (arrjon) [reply pypesto/visualize/model_fit.py:79]: None, I will change that

**Review by Doresic — APPROVED** [review]
Looks good to me! Thanks
Might be good to add a test in `test/test_visualize.py` testing a case with an aggregated objective (similar to `test_visualize_optimized_model_fit()` or an extension of it).

**Review by PaulJonasJost — APPROVED** [review]

**Review by stephanmg — APPROVED** [review]

**Pull request merged**